### ISSUE TITLE
Allow ES6 class generator methods

### ISSUE THREAD

**Issue by elierotenberg** [jshint/jshint#1978]
Using jshint 2.5.10 with option `esnext: true`, the following code:
```js
class A {
  *x() {
    return yield void 0;
  }
}
```
yields:
```
line 2: Unexpected '*'.
line 3: A yield statement shall be within a generator function (with syntax: 'function*')
```
Oddly enough, though, the following code yields no error:
```js
let A = { *x() { return yield void 0; } };
```

I believe there may be a disrupting interaction between classes and generators, preventing jshint to acknowledge `*x()` in the class definition of symbol `A` as semantically desugaring to `A.prototype.x = function*() { ... }`.
For reference, as I failed to find many examples out there with generator class methods, see [the strawman](http://wiki.ecmascript.org/doku.php?id=strawman:maximally_minimal_classes) and [the spec](https://people.mozilla.org/~jorendorff/es6-draft.html#sec-method-definitions)).

**Comment by baileyparker** [comment]
Should this be closed? As far as I can tell this was fixed in 2.8.

**Comment by nicolo-ribaudo** [comment]
Fixed - #2066
